app/uk/gov/hmrc/customs/emailfrontend/controllers/EmailConfirmedController.scala

```diff
@@ -40,46 +39,32 @@ class EmailConfirmedController @Inject()(actions: Actions, view: email_confirmed
                                          updateVerifiedEmailService: UpdateVerifiedEmailService,
                                          mcc: MessagesControllerComponents)
                                         (implicit override val messagesApi: MessagesApi, ec: ExecutionContext)
-  extends FrontendController(mcc) with I18nSupport with DetermineRouteController {
+  extends FrontendController(mcc) with I18nSupport {
 
   def show: Action[AnyContent] = (actions.authEnrolled andThen actions.isPermitted andThen actions.eori).async { implicit request =>
-    for{
-      status <- emailCacheService.emailAmendmentStatus(request.user.internalId)
-      result <- redirectBasedOnAmendmentStatus(status)(redirectBasedOnEmailStatus)
-    } yield {
-      result
-    }
-
+     emailCacheService.emailAmendmentData(request.user.internalId)(redirectBasedOnEmailStatus)
   }
 
-  private def redirectBasedOnEmailStatus(implicit request: EoriRequest[AnyContent]): Future[Result] ={
-    emailCacheService.fetchEmail(request.user.internalId) flatMap {
-      _.fold {
-        Logger.warn("[EmailConfirmedController][show] - emailStatus cache none, user logged out")
-        Future.successful(Redirect(SignOutController.signOut()))
-      } {
-        emailStatus =>
-          for {
-            verified <- emailVerificationService.isEmailVerified(emailStatus.email)
-            redirect <- if (verified.contains(true)) updateEmail(emailStatus) else Future.successful(Redirect(VerifyYourEmailController.show()))
-          } yield redirect
-      }
-    }
+  private def redirectBasedOnEmailStatus(email: String)(implicit request: EoriRequest[AnyContent]): Future[Result] = {
+    for {
```

Comment:
I still struggle to understand why you guys like for comprehension in situations like that (2 futures) :) I'd still make it with map

Reply:
not intended to change this for now, it's existing